lib/inspec/backend.rb

```diff
@@ -53,6 +53,14 @@ def self.create(config) # rubocop:disable Metrics/AbcSize
         raise "Can't connect to transport backend '#{name}'."
       end
 
+      # Cisco IOS requires a special implementation of `Net:SSH`. This uses the
+      # SSH transport to identify the platform, but then replaces that transport
+      # with the CiscoIOS transport in order to behave as expected later.
+      if name == 'ssh' && connection.platform.cisco_ios?
```

Comment:
This bothers me a bit, as we use the ssh connection to run commands to see what platform its on and then override the platform. I think we should just have the user use `cisco_ios` transport when needed.

Reply:
Feels like we have some conceptual parts that we need to discuss about the cisco specific transport